cfme/physical/physical_switch.py

```diff
@@ -0,0 +1,238 @@
+from cfme.utils.wait import wait_for
+
+
+@attr.s
+class PhysicalSwitch(BaseEntity, Updateable, Pretty, PolicyProfileAssignable, Taggable):
+    """Model of an Physical Switch in cfme.
+
+    Args:
+        name: Name of the physical Switch.
+
+    Usage:
+
+        myswitch = PhysicalSwitch(name='sample_switch')
+        myswitch.create()
+
+    """
+    pretty_attrs = ['name']
+
+    name = attr.ib()
+    provider = attr.ib(default=None)
+    db_id = None
+
+    INVENTORY_TO_MATCH = ['power_state', 'part_number', 'serial_number', 'description',
+                          'manufacturer']
+
+    def load_details(self, refresh=False):
+        """To be compatible with the Taggable and PolicyProfileAssignable mixins.
+
+        Args:
+            refresh (bool): Whether to perform the page refresh, defaults to False
+        """
+        view = navigate_to(self, "Details")
+        if refresh:
+            view.browser.refresh()
+            view.flush_widget_cache()
+
+    def _execute_button(self, button, option, handle_alert=False):
+        view = navigate_to(self, "Details")
+        view.toolbar.custom_button(button).item_select(option, handle_alert=handle_alert)
+        return view
+
+    def _execute_action_button(self, button, option, handle_alert=True, **kwargs):
+        target = kwargs.get("target", None)
+        provider = kwargs.get("provider", None)
+        desired_state = kwargs.get("desired_state", None)
+
+        view = self._execute_button(button, option, handle_alert=handle_alert)
+
+        if desired_state:
+            self._wait_for_state_change(desired_state, target, provider, view)
+        elif handle_alert:
+            wait_for(
+                lambda: view.flash.is_displayed,
+                message="Wait for the handle alert to appear...",
+                num_sec=5,
+                delay=2
+            )
+
+    @variable(alias='ui')
+    def power_state(self):
+        view = navigate_to(self, "Details")
+        return view.entities.power_management.get_text_of("Power State")
+
+    @variable(alias='ui')
+    def part_number(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Part Number")
+
+    @variable(alias='ui')
+    def serial_number(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Serial Number")
+
+    @variable(alias='ui')
+    def description(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Description")
+
+    @variable(alias='ui')
+    def manufacturer(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Manufacturer")
+
+    def _wait_for_state_change(self, desired_state, target, provider, view, timeout=300, delay=10):
+        """Wait for PhysicalSwitch to come to desired state. This function waits just the needed amount of
+           time thanks to wait_for.
```

Comment:
**REQUIRED**
Please fix that docstring.

Reply:
Done.